src/NHibernate/Engine/EntityEntry.cs

```diff
@@ -205,6 +205,7 @@ public EntityKey EntityKey
 
 		public object GetLoadedValue(string propertyName)
 		{
+			if (loadedState == null) return null;
```

Comment:
This doesn't look like correct fix. You should really investigate why `entityEntry.LoadedState` is null in your case. It looks like the real issue in some other place.

Maybe in throwing place value should be retrieved as:
```C#
entityEntry.LoadedState != null
				? entityEntry.GetLoadedValue(propertyName)
				: entityEntry.Persister.GetPropertyValue(owner, propertyName);
```

But it's really hard to say without unit test.

Reply:
LoadedState can be null for readonly entities. However I don’t know the exact reason why it is null here. We definitely need a test case.

Reply:
[](https://github.com/nhibernate/nhibernate-core/issues/2137)
I have now a test code, sorry for issue duplicate, first time contributing on github.

Reply:
https://github.com/nhibernate/nhibernate-core/issues/2137

Reply:
Created a new pull request, with unit test and fix as suggested by @bahusoid